### ISSUE TITLE
[json] Inline comment of last command doesn't stay in place when adding new setting

### ISSUE THREAD

**Issue by GandalfSaxe** [microsoft/vscode#56388]
Issue Type: <b>Bug</b>

The comment on the last setting doesn't stay put, but is moved down to the new command, which is not ideal:

![](https://i.imgur.com/OtgBnWZ.gif)

VS Code version: Code - Insiders 1.27.0-insider (8727085fff33bf7d8f5e27483b10f72508d4167a, 2018-08-14T05:12:11.579Z)
OS version: Darwin x64 18.0.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-4870HQ CPU @ 2.50GHz (8 x 2500)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: unavailable_software<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|9, 7, 6|
|Memory (System)|16.00GB (0.56GB free)|
|Process Argv|/Users/gandalf/Documents/portable/Visual Studio Code - Insiders.app/Contents/MacOS/Electron|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (34)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|9.0.3
spellright|ban|2.5.30
path-intellisense|chr|1.4.2
bracket-pair-colorizer|Coe|1.0.59
vscode-svgviewer|css|1.4.4
vscode-markdownlint|Dav|0.19.0
githistory|don|0.4.2
jupyter|don|1.1.4
vscode-pandoc|Dou|0.0.8
gitlens|eam|8.5.4
vsc-material-theme|Equ|2.3.0
prettier-vscode|esb|1.6.1
vscode-diff|fab|1.3.3
wolfram-language|fli|0.1.0
latex-workshop|Jam|5.7.0
python|ms-|2018.7.1
PowerShell|ms-|1.8.2
wordcount|ms-|0.1.0
vsliveshare|ms-|0.3.535
gremlins|nho|0.7.0
vscode-docker|Pet|0.1.0
quicktype|qui|12.0.36
vscode-icons|rob|7.25.0
partial-diff|ryu|1.3.0
markdown-preview-enhanced|shd|0.3.5
vscode-spotify|shy|2.5.1
vscode-fileutils|sle|2.10.3
slack|soz|0.0.14
code-spell-checker|str|1.6.10
pdf|tom|0.3.0
vscode-terminal-here|Tyr|0.1.1
vscodeintellicode|Vis|1.0.3
vscode-todo-highlight|way|1.0.4
markdown-all-in-one|yzh|1.6.0

(2 theme extensions excluded)

</details>

**Comment by sandy081** [comment]
Closing this in favour of the new settings editor.

**Comment by GandalfSaxe** [comment]
Is the new editor going to *completely* replace .json file view? Otherwise it seems like a bug worth fixing.

**Comment by sandy081** [comment]
Yes, New editor is supposed to replace this mixed UI editor. Opening just the json file without these actions will be supported.

@roblourens FYI

**Comment by GandalfSaxe** [comment]
But it's still a problem. I didn't show in the original post, but it happens no matter how the settings are added to the json. If I to into the new editor and change a setting, then the comment on the last line in `settings.json` jumps always down to the newly added setting. It's not a problem with the mixed UI, it's a fundamental problem with how new settings are added, taking the comments on the last line with them, regardless of how they are added. 

In other words, you couldn't reliably add comments to your `settings.json`, even if you only use the new settings UI in the future + open and comment the `settings.json` separately. The last comment is always gonna be unreliably jumping around.

**Comment by roblourens** [comment]
@sandy081 this isn't specific to the mixed UI editor. It also happens when editing with the new editor, or I assume when an extension or something else changes a setting. So it's a problem with the service that writes config changes to JSON.

**Comment by sandy081** [comment]
Agreed

**Comment by sandy081** [comment]
@aeschli I use json edit to get the positions. Can the json edit consider the comments and provide positions accordingly.

**Comment by aeschli** [comment]
duplicate of https://github.com/microsoft/vscode/issues/24676
